nucliadb/src/nucliadb/search/search/query.py

```diff
@@ -382,12 +409,19 @@ async def parse_min_score(self, request: nodereader_pb2.SearchRequest, incomplet
             semantic_min_score = self.min_score.semantic
         elif self.has_vector_search and not incomplete:
             query_information = await self._get_query_information()
-            if query_information.semantic_threshold is not None:
-                semantic_min_score = query_information.semantic_threshold
+            vectorset = await self._select_vectorset()
+            if vectorset is not None:
+                semantic_threshold = query_information.semantic_thresholds.get(vectorset, None)
+                if semantic_threshold is not None:
+                    semantic_min_score = semantic_threshold
+                else:
+                    logger.warning(
+                        "Semantic threshold not found in query information, using default",
+                        extra={"kbid": self.kbid},
+                    )
             else:
                 logger.warning(
-                    "Semantic threshold not found in query information, using default",
-                    extra={"kbid": self.kbid},
+                    "Vectorset unset, using default semantic threshold", extra={"kbid": self.kbid}
```

Comment:
will this log be very chatty? I assume lots of requests won't have the vectorset param provided and will just use the default no?

Reply:
No, this is a log for either predict or the user setting the vectorset. Shouldn't happen. I'll clarify the log